src/widget/sidebar/extra.c

```diff
@@ -385,44 +394,100 @@ static int draw_request_buttons(int y_offset)
+            int width = data.x_offset + 10;
+
+            int image_id = image_group(GROUP_RESOURCE_ICONS) + RESOURCE_WEAPONS;
+            const image *img = image_get(image_id);
+            int image_y_offset = (EXTRA_INFO_LINE_SPACE - img->height) / 2;
+
+            image_draw(image_id, width, y_offset + image_y_offset - 2);
+
             int force_text_offset = get_text_offset_for_force_size(r->amount);
+
             text_draw_ellipsized(translation_for(TR_SIDEBAR_EXTRA_REQUESTS_SMALL_FORCE + force_text_offset),
-                data.x_offset + 32, y_offset, data.width - 34, FONT_NORMAL_WHITE, 0);
-            y_offset += EXTRA_INFO_LINE_SPACE;
-            lang_text_draw_amount(8, 4, r->time, data.x_offset + 26, y_offset + 4,
-                r->time <= 3 ? FONT_NORMAL_RED : FONT_NORMAL_WHITE);
+                data.x_offset + 32, y_offset - 7, data.width - 34, FONT_NORMAL_GREEN, 0);
+
+            lang_text_draw_amount(8, 4, r->time, data.x_offset + 26, y_offset + 6,
+                r->time <= 3 ? FONT_NORMAL_RED : FONT_NORMAL_GREEN);
+
+            text_draw_centered(translation_for(TR_SIDEBAR_EXTRA_REQUESTS_SEND),
+                               data.x_offset + 2 , y_offset + 25, 158, FONT_NORMAL_GREEN, 0);
+
         } else {
             int width = data.x_offset + 10;
-            width += text_draw_number(r->amount, 0, "",
-                width, y_offset, FONT_NORMAL_WHITE);
+
             int resource_offset = r->resource + resource_image_offset(r->resource, RESOURCE_IMAGE_ICON);
+
             int image_id = image_group(GROUP_RESOURCE_ICONS) + resource_offset;
             const image *img = image_get(image_id);
             int image_y_offset = (EXTRA_INFO_LINE_SPACE - img->height) / 2;
+
             image_draw(image_id, width, y_offset + image_y_offset);
+
+            width += img->width + 6;
+
             if (r->resource != RESOURCE_DENARII) {
-                width += img->width + 6;
-                width += text_draw(string_from_ascii("("), width, y_offset, FONT_NORMAL_WHITE, 0) - 6;
-                width += text_draw(translation_for(TR_SIDEBAR_EXTRA_REQUESTS_HAVE),
-                    width, y_offset, FONT_NORMAL_WHITE, 0);
-                text_draw_number(r->available, 0, ")", width, y_offset, FONT_NORMAL_WHITE);
+                int is_stockpiled = city_resource_is_stockpiled(r->resource);
+                int enough_resource = 0;
+
+                int request_index = data.requests[i].index;
+                if (city_request_has_troop_request() && i != 0) {
+                    request_index++;
+                }
+
+                int status = city_request_get_status(request_index);
+
+                // button text
+                if (status) {
+                    if (status == CITY_REQUEST_STATUS_NOT_ENOUGH_RESOURCES) {
+                        if (is_stockpiled) {
+                            image_draw(assets_get_image_id(assets_get_group_id("Areldir", "UI_Elements"),
```

Comment:
I know I suggested a visual feedback when goods were being stockpiled, but I was kinda dumb when I made that suggestion.

If goods are stockpiled, the button already displays "stop stockpiling"! So this small icon serves no real purpose...

Reply:
The icon add a feedback to the user. Personally i like this feedback, it's more clear than only text on button.

Reply:
We can keep it, sure!